File: app/src/main/java/com/diwixis/filmlibrary/movies_module/MovieItemAdapter.java
package com.diwixis.filmlibrary.movies_module;


import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;

import com.bumptech.glide.Glide;
import com.diwixis.filmlibrary.R;
import com.diwixis.filmlibrary.api.Urls;
import com.diwixis.filmlibrary.structures.Result;

import java.util.ArrayList;
import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;

/**
 * Created by Diwixis on 18.04.2017.
 */

class MovieItemAdapter extends RecyclerView.Adapter<MovieItemAdapter.ViewHolder> {

    private List<Result> movieList = new ArrayList<>();
    private IOnItemClick clickListener = null;
    private int width = 0;

    void setData(List<Result> list, int width){
        movieList = list;
        notifyDataSetChanged();
        this.width = width;
    }

    void setClickListener(IOnItemClick listener){
        clickListener = listener;
    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_movie, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(ViewHolder holder, int position) {
        Result movie = movieList.get(position);
        Glide.with(holder.image.getContext())
                .load(Urls.IMAGE_URL+movie.getPosterPath())
                .asBitmap()
                .override(width, width)
                .centerCrop()
                .into(holder.image);
    }

    @Override
    public int getItemCount() {
        return movieList == null ? 0 : movieList.size();
    }

    public interface IOnItemClick{
        void onItemClick(Result result);
    }

    class ViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener{
        @BindView(R.id.image)
        ImageView image;

        public ViewHolder(View itemView) {
            super(itemView);
            ButterKnife.bind(this, itemView);
            itemView.setOnClickListener(this);
            image.requestLayout();
        }

        @Override
        public void onClick(View v) {
            clickListener.onItemClick(movieList.get(getAdapterPosition()));
        }
    }
}

File: app/src/main/java/com/diwixis/filmlibrary/movies_module/MovieItemAdapter.kt
package com.diwixis.filmlibrary.movies_module

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.diwixis.filmlibrary.R
import com.diwixis.filmlibrary.api.Urls
import com.diwixis.filmlibrary.data.Result
import kotlinx.android.synthetic.main.item_movie.view.*
import java.util.*

/**
 * Created by Diwixis on 18.04.2017.
 */
internal class MovieItemAdapter :
    RecyclerView.Adapter<MovieItemAdapter.ViewHolder>() {
    private var movieList: List<Result>? = ArrayList()
    private var clickListener: IOnItemClick? = null
    private var width = 0
    fun setData(
        list: List<Result>?,
        width: Int
    ) {
        movieList = list
        notifyDataSetChanged()
        this.width = width
    }

    fun setClickListener(listener: IOnItemClick?) {
        clickListener = listener
    }

    override fun onCreateViewHolder(
        parent: ViewGroup,
        viewType: Int
    ): ViewHolder {
        val view =
            LayoutInflater.from(parent.context).inflate(R.layout.item_movie, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(
        holder: ViewHolder,
        position: Int
    ) {
        holder.bind(movieList!![position])
    }

    override fun getItemCount(): Int {
        return if (movieList == null) 0 else movieList!!.size
    }

    interface IOnItemClick {
        fun onItemClick(result: Result?)
    }

    internal inner class ViewHolder(itemView: View) :
        RecyclerView.ViewHolder(itemView), View.OnClickListener {
        override fun onClick(v: View) {
            clickListener!!.onItemClick(movieList!![adapterPosition])
        }

        fun bind(movie: Result) {
            itemView.setOnClickListener(this)
            Glide.with(itemView.image.context)
                .load(Urls.IMAGE_URL + movie.posterPath)
                .override(width, width)
                .centerCrop()
                .into(itemView.image)
            itemView.image.requestLayout()
        }
    }
}